### PR TITLE
Add option to disable xrdcp write recovery - wmagent branch

### PR DESCRIPTION
**Pull request by amaltaro**
wmagent version of #7887
Not exactly the same, since we support both old and new options here.
This one is also worth a review, Dirk :-)

### CODE REVIEW

File: src/python/WMCore/Storage/Backends/XRDCPImpl.py [hunk 1]
```diff
@@ -62,8 +62,15 @@ def createStageOutCommand(self, sourcePFN, targetPFN, options=None, checksums=No
         parser = argparse.ArgumentParser()
         parser.add_argument('--cerncastor', action='store_true')
         parser.add_argument('--old', action='store_true')
+        parser.add_argument('--disablewriterecovery', action='store_true')
```
Comment (hufnagel) [inline src/python/WMCore/Storage/Backends/XRDCPImpl.py]: Why ? We don't have any old-style syntax in site-local-config to worry about for this parameter.

Comment (hufnagel) [inline src/python/WMCore/Storage/Backends/XRDCPImpl.py]: And no workflows (LogCollect) that set that either.

File: src/python/WMCore/Storage/Backends/XRDCPImpl.py [hunk 2]
```diff
@@ -111,6 +118,10 @@ def createStageOutCommand(self, sourcePFN, targetPFN, options=None, checksums=No
                     for initFile in initFiles:
                         copyCommand += "source %s\n" % initFile
 
+        if (args.wma_disablewriterecovery or args.disablewriterecovery) and not\
+                (args.wma_cerncastor or args.cerncastor):
```
Comment (hufnagel) [inline src/python/WMCore/Storage/Backends/XRDCPImpl.py]: Same comment as in the other PR, we can just remove that secondary check, cleaner that way.

File: src/python/WMCore/Storage/Backends/XRDCPImpl.py [hunk 3]
```diff
@@ -62,8 +62,15 @@ def createStageOutCommand(self, sourcePFN, targetPFN, options=None, checksums=No
...
         parser.add_argument('--old', action='store_true')
+        parser.add_argument('--disablewriterecovery', action='store_true')
+        parser.add_argument('--wma-cerncastor', action='store_true')
+        parser.add_argument('--wma-old', action='store_true')
+        parser.add_argument('--wma-disablewriterecovery', action='store_true')
         args, unknown = parser.parse_known_args(options.split())
 
+        # strip out WMAgent specific options
+        unknown = set(unknown) - {'--cerncastor', '--old', '--disablewriterecovery',
+                                  '--wma-cerncastor', '--wma-old', '--wma-disablewriterecovery'}
```
Comment (hufnagel) [inline src/python/WMCore/Storage/Backends/XRDCPImpl.py]: Any reason you don't use the more general removal of generic wma options here ?

**Review by hufnagel — CHANGES_REQUESTED** [review]

**Comment by amaltaro** [comment]
Thanks, Dirk. Code is updated according to your suggestions.

**Comment by amaltaro** [comment]
[Fix #7888 - Add option to disable xrdcp write recovery; remove xrdcp-old](https://trello.com/c/fmWpPxyZ/36-fix-%237888---add-option-to-disable-xrdcp-write-recovery%3B-remove-xrdcp-old)

**Comment by amaltaro** [comment]
My tests were successful. I'm going to push it to production agents ASAP such that @hufnagel  can update the site-local-config, before his day is over. Otherwise I leave these changes to a Friday, which is far from ideal.

**Comment by amaltaro** [comment]
Dirk, all the production agents (+ relval + hepcloud) were just patched.

**Comment by hufnagel** [comment]
Yeah, but the Tier0 hasn't been patched yet, right ? So as is I can update the SITECONF for T2_CH_CERN but not T0_CH_CERN.

**Comment by amaltaro** [comment]
Correct, not the T0 node(s). Actually, I usually don't touch T0 agents. Can you patch it and make sure it's used for any future replays?

**Pull request merged**